dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -129,46 +131,73 @@ public static void main(String[] arguments)
   }
 
   /**
-   * Finds jar file paths for Maven artifacts and their dependencies.
+   * Finds jar file paths for Maven artifacts and their transitive dependencies.
    *
    * @param artifacts Maven artifacts to check
    * @return list of absolute paths to jar files
    * @throws RepositoryException when there is a problem in retrieving jar files
    */
   public static ImmutableList<Path> artifactsToClasspath(List<Artifact> artifacts)
       throws RepositoryException {
-    
+
     LinkedListMultimap<Path, DependencyPath> multimap = artifactsToPaths(artifacts);
     return ImmutableList.copyOf(multimap.keySet());
   }
-  
-  
+
   // Multimap is a pain, maybe just use LinkedHashMap<Path, List<DependencyPath>>
   /**
-   * Finds jar file paths for Maven artifacts and their dependencies.
+   * Finds jar file paths and dependency paths for Maven artifacts and their transitive
+   * dependencies. When there are multiple versions of an artifact, the closest to the root ({@code
+   * artifacts}) is picked up into the key of the return value. This 'pick closest' strategy follows
```

Comment:
drop "into the key of the return value". The issue here are the values, not the keys. Can you elaborate why the values are a list instead of a single path? That is, why do we need a multimap?

Reply:
Dropped unnecessary phrase, and added "The values of the returned map for a key (jar file) are the different Maven dependency paths from ({@code artifacts}) to the Maven artifact of the jar file."

Multimap gives more information. For example:

```
grpc-core-1.17.1.jar (4 errors):
  Linked from: io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-alts:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-auth:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-grpclb:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-netty:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-netty-shaded:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-okhttp:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-protobuf:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-protobuf-lite:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-protobuf-nano:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-stub:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-testing:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-alts:1.17.1 / io.grpc:grpc-auth:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-alts:1.17.1 / io.grpc:grpc-netty-shaded:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-alts:1.17.1 / io.grpc:grpc-protobuf:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-alts:1.17.1 / io.grpc:grpc-stub:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-alts:1.17.1 / io.grpc:grpc-grpclb:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-grpclb:1.17.1 / io.grpc:grpc-protobuf:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-grpclb:1.17.1 / io.grpc:grpc-stub:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-protobuf:1.17.1 / io.grpc:grpc-protobuf-lite:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-testing:1.17.1 / io.grpc:grpc-stub:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-alts:1.17.1 / io.grpc:grpc-protobuf:1.17.1 / io.grpc:grpc-protobuf-lite:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-alts:1.17.1 / io.grpc:grpc-grpclb:1.17.1 / io.grpc:grpc-protobuf:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-alts:1.17.1 / io.grpc:grpc-grpclb:1.17.1 / io.grpc:grpc-stub:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-grpclb:1.17.1 / io.grpc:grpc-protobuf:1.17.1 / io.grpc:grpc-protobuf-lite:1.17.1 / io.grpc:grpc-core:1.17.1
  Linked from: io.grpc:grpc-alts:1.17.1 / io.grpc:grpc-grpclb:1.17.1 / io.grpc:grpc-protobuf:1.17.1 / io.grpc:grpc-protobuf-lite:1.17.1 / io.grpc:grpc-core:1.17.1
  MethodSymbolReference{sourceClassName=io.grpc.internal.ServiceConfigInterceptor$1DelayedHedgingPolicyProvider, targetClassName=com.google.common.base.Verify, methodName=verify, descriptor=(ZLjava/lang/String;Ljava/lang/Object;)V}, reason: SYMBOL_NOT_FOUND, target class from /usr/local/google/home/suztomo/.m2/repository/com/google/guava/guava/20.0/guava-20.0.jar
```
is more informative than

```
grpc-core-1.17.1.jar (4 errors):
  Linked from: io.grpc:grpc-core:1.17.1
MethodSymbolReference{sourceClassName=io.grpc.internal.ServiceConfigInterceptor$1DelayedHedgingPolicyProvider, targetClassName=com.google.common.base.Verify, methodName=verify, descriptor=(ZLjava/lang/String;Ljava/lang/Object;)V}, reason: SYMBOL_NOT_FOUND, target class from /usr/local/google/home/suztomo/.m2/repository/com/google/guava/guava/20.0/guava-20.0.jar
```